packages/backend-data/src/convert_authorization_modes.ts

```diff
@@ -95,13 +95,34 @@ const convertOIDCAuthConfigToCDK = ({
 
 /**
  * Compute default auth mode based on availability of auth resources.
- * Will specify userPool if both user pool is specified, and no auth resources are specified, else rely on override value if needed.
+ *
+ * Will specify userPool if both user pool is specified, and no auth resources are specified.
+ *
+ * If no auth resources are provided and there's only one mode we'll use that as we implicitly always add iam auth.
  */
 const computeDefaultAuthorizationMode = (
   providedAuthConfig: ProvidedAuthConfig | undefined,
   authModes: AuthorizationModes | undefined
 ): DefaultAuthorizationMode | undefined => {
-  if (providedAuthConfig && !authModes) return 'userPool';
+  if (!providedAuthConfig && !authModes) {
+    // if no auth is configured or provided use api key.
+    // see also computeApiKeyAuthFromResource
+    return 'apiKey';
+  } else if (providedAuthConfig && !authModes) {
+    return 'userPool';
+  } else if (
+    !providedAuthConfig &&
+    authModes &&
+    Object.keys(authModes).length === 1
+  ) {
+    if (authModes.oidcAuthorizationMode) {
+      return 'oidc';
+    } else if (authModes.lambdaAuthorizationMode) {
+      return 'lambda';
+    } else if (authModes.apiKeyAuthorizationMode) {
+      return 'apiKey';
+    }
+  }
   return;
 };
```

Comment:
Should the fallback be an error?

Reply:
The way this used to work is - when this defaults to `undefined` it falls back to data construct validator which complains about missing default.

I think it's not a bad idea to throw here instead - so that error message corresponds to `defineData` API and not construct's which might be confusing to interpret.

Reply:
I think that's a good idea to make it more obvious that undefined is a failure case.